README.md

```diff
@@ -177,6 +177,57 @@ storage.directory=/var/lib/janusgraph/data
 index.search.directory=/var/lib/janusgraph/index
 ```
 
+#### Gremlin Server Environment Variable Syntax
+Environment Variables that start with the prefix `gremlinserver.*` or `gremlinserver%d.*` are used to edit the base gremlin-server.yaml file. The text after the prefix in the environment variable name should follow a specific syntax. This syntax is implemented using the [yq][YQ_GITHUB] write and delete commands and the [yq documentation][YQ_DOC] can be used as a reference for this syntax. Secondly, the value of the environment variable will be used to set the value of the key specified in the environment variable name.
+
+Let's take a look at a few examples:
+
+##### Nested Properties
+
+```text
+$ docker run --rm -it -e gremlinserver.graphs.ConfigurationManagementGraph=conf/JanusGraph-configurationmanagement.properties janusgraph/janusgraph:0.4.0 janusgraph show-config
+...
+graphs:
+  graph: conf/gremlin-server/janusgraph-cql-es-server.properties
+  ConfigurationManagementGraph: conf/JanusGraph-configurationmanagement.properties
+scriptEngines:
+...
+```
+##### Delete a component
+To delete a component append %d to the 'gremlinserver.' prefix before the closing dot and then select the component following the prefix. Don't forget the trailing '='.
+
+```text
+$ docker run --rm -it -e gremlinserver%d.graphs.graph= janusgraph/janusgraph:0.4.0 janusgraph show-config
+...
+channelizer: org.apache.tinkerpop.gremlin.server.channel.WebSocketChannelizer
+graphs: {}
+scriptEngines:
+...
+```
+
+##### Append item and alternate indexing syntax
```

Comment:
I don't understand this header. How is the indexing syntax alternated? Maybe a sentence that introduces the code example could also help.

Reply:
By alternate indexing syntax I mean using square brackets instead of dot notation. e.g. p.name vs p["name"] or in the example provided in the README file I have `plugins["org.apache.tinkerpop.gremlin.jsr223.ScriptFileGremlinPlugin"]`. In this case you actually have to use the square bracket syntax since there are dots in the name. 

I agree a description below the header would be helpful. See below this paragraph for my attempt at a description. Does the header make sense now with the added description? Is alternate indexing the right words here? Some people say indexing some people say dereferencing when talking about the action of using the key to get it's associated value in a map. What do you think is a good header for this section? Is the below description good or would you add anything?

```
##### Append item and alternate indexing syntax
+ This example shows how to append an item to a list and also shows an example of the square bracket syntax which can also be used to reference a key in a path that contains special characters.

Reply:
This descriptions sounds good to me. Please only note that long lines should be wrapped (100 characters per line max) and there should be a newline after a heading.